src/query/functions/fetch.go

```diff
@@ -103,23 +106,32 @@ func (n *FetchNode) fetch(queryCtx *models.QueryContext) (block.Result, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "fetch")
 	defer sp.Finish()
 
-	timeSpec := n.timespec
-	// No need to adjust start and ends since physical plan
-	// already considers the offset, range
-	startTime := timeSpec.Start
-	endTime := timeSpec.End
+	var (
+		timeSpec = n.timespec
+		// No need to adjust start and ends since physical plan already considers
+		// the offset, range.
+		startTime = timeSpec.Start
+		endTime   = timeSpec.End
+		step      = timeSpec.Step
+		offset    = n.op.Offset
+	)
+
+	if n.overrideOptions.ShouldOverride {
+		step = n.overrideOptions.Step
```

Comment:
Hm, should step only be applied if `n.overrideOptions.Step != 0`? Seems like that is special cased in the lazy transform (only use override step if override step is set to some value) but is not special cased here possibly? 

I guess I'm not entirely clued into when step is overridden, etc.

Reply:
So this is to [mirror Prom](https://prometheus.io/blog/2019/01/28/subquery-support/), which allows the Step here to be omitted and defaults to the default query step.

Reply:
I see, sounds good.